homeworks/vancsikpeti/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,114 @@
+class Car:
+    total_distance = 0.0
+
+    def __init__(self, brand: str, model: str, year: int, mileage = 0, fuel_level = 100):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = mileage # induló értéke 0
+        self.fuel_level = fuel_level # induló értéke 100, tételezzük fel, hogy 0.1% üzemanyag fogy megtett kilométerenként
+   
+    #def new_car(self):
+        #self.brand = input("Brand of car: ")
+        #self.model = input("Model of car: ")
+        #self.year = input("Product year: ")
+        #self.fuel_level = 100
+        #self.mileage = 0
+
+    def show_cars_condition(self):
+        print(f"{self.brand}-{self.model} ({self.year}): distance->{self.mileage} miles, fuel level->{self.fuel_level}%")
+    
+    def __str__(self):
+        return f"{self.brand}-{self.model} ({self.year})"
+    
+    def enter_miles(self):
```

Comment:
itt két esetben is ugyanazt printeled ki: `print("Please enter a positive whole number.")`.
Ezt le tudnád egyszerűsíteni, ha a try-except-ben lekezelnéd a negatív értékeket is:

```
    def enter_miles(self):
        while True:
            try: 
                miles = int(input(f"How many miles did you drive with {self}? "))
                if miles < 0:
                    raise ValueError
            except (ValueError,TypeError):
                print("Please enter a positive whole number.")
                continue
            # fuel check
            if miles * 0.1 > self.fuel_level:
                print(f"Not enough fuel in {self}. The fuel level is {self.fuel_level}%.")
            else: 
                return miles
```

Illetve az else-ben igazából egyből return-ölheted a miles-t, nem kell break-elni a loop-ból hogy aztán return-ölj.